blocklib/timing/test/qa_timingSource.cpp

```diff
@@ -0,0 +1,221 @@
+#include <TimingSource.hpp>
+#include <boost/ut.hpp>
+#include <fmt/core.h>
+#include <gnuradio-4.0/Scheduler.hpp>
+#include <gnuradio-4.0/basic/ConverterBlocks.hpp>
+#include <gnuradio-4.0/testing/ImChartMonitor.hpp>
+#include <gnuradio-4.0/testing/TagMonitors.hpp>
+#include <latch>
+
+using namespace std::string_literals;
+using namespace std::chrono_literals;
+using namespace boost::ut;
+
+namespace fair::timing::test {
+
+const suite TimingBlockHelpers = [] {
+    "filter-parsing"_test = []() {
+        auto expectTiming = [](auto a, auto b) {
+            auto aa = gr::timing::TimingSource::parseFilter(a);
+            expect(aa == b) << fmt::format("checking: {}\ngot: {:#x}, {:#x}\nexp: {:#x}, {:#x}\n", a, aa.first, aa.second, b.first, b.second);
+        };
+
+        expectTiming("", std::pair{0x1000000000000000ul, 0xf000000000000000ul});
+        expectTiming("SIS100_RING:CMD_BP_START", std::pair{0x1136100000000000ul, 0xfffffff000000000ul});
+        expectTiming("SIS100_RING:CMD_BP_START:BEAM-IN=1", std::pair{0x1136100800000000ul, 0xfffffff800000000ul});
+        expectTiming("SIS100_RING:CMD_BP_START:BEAM-IN=1:BPC-START=0", std::pair{0x1136100800000000ul, 0xfffffffc00000000ul});
+        expectTiming("SIS100_RING:CMD_BP_START:BEAM-IN=1:BPC-START=0:0", std::pair{0x1136100800000000ul, 0xfffffffffff00000ul});
+        expectTiming("SIS100_RING:CMD_BP_START:BEAM-IN=1:BPC-START=0:0:2", std::pair{0x1136100800000080ul, 0xffffffffffffffc0ul});
+        expectTiming("301:256", std::pair{0x112d100000000000ul, 0xfffffff000000000ul});
```

Comment:
Please consider to add not-correct string: without `SIS100_RING`, `CMD_BP_START`, or some incorrect names.